src/main/java/frc/robot/subsystems/scoring/AimerIOSim.java

```diff
@@ -3,31 +3,75 @@
 import edu.wpi.first.math.controller.ArmFeedforward;
 import edu.wpi.first.math.controller.PIDController;
 import edu.wpi.first.math.system.plant.DCMotor;
+import edu.wpi.first.math.trajectory.TrapezoidProfile;
+import edu.wpi.first.math.trajectory.TrapezoidProfile.State;
+import edu.wpi.first.wpilibj.Timer;
 import edu.wpi.first.wpilibj.simulation.SingleJointedArmSim;
 import frc.robot.Constants;
-import frc.robot.Constants.Scoring;
+import frc.robot.Constants.ScoringConstants;
 
 public class AimerIOSim implements AimerIO {
     // TODO: Tune this later
-    private final SingleJointedArmSim sim = new SingleJointedArmSim(DCMotor.getNeoVortex(2), 1.0, 1.0, 1.0, 0.0, 1.0, true, 0.0);
-    private final PIDController controller = new PIDController(Scoring.aimerkP, Scoring.aimerkI, Scoring.aimerkD);
-    private final ArmFeedforward feedforward = new ArmFeedforward(Scoring.aimerkS, Scoring.aimerkG, Scoring.aimerkV, Scoring.aimerkA);
+    private final SingleJointedArmSim sim =
+            new SingleJointedArmSim(
+                    DCMotor.getNeoVortex(2),
+                    1.0,
+                    SingleJointedArmSim.estimateMOI(0.5, 4.5),
+                    0.5,
+                    0.0,
+                    2.0,
+                    false,
+                    0.0);
+    private final PIDController controller =
+            new PIDController(
+                    ScoringConstants.aimerkP, ScoringConstants.aimerkI, ScoringConstants.aimerkD);
+    private final ArmFeedforward feedforward =
+            new ArmFeedforward(
+                    ScoringConstants.aimerkS,
+                    ScoringConstants.aimerkG,
+                    ScoringConstants.aimerkV,
+                    ScoringConstants.aimerkA);
+    private final TrapezoidProfile profile =
+            new TrapezoidProfile(new TrapezoidProfile.Constraints(0.8, 0.4));
 
-    double goalAngRad = 0.0;
+    private final Timer timer = new Timer();
+
+    double goalAngleRad = 0.0;
     double appliedVolts = 0.0;
 
+    double initialAngle = 0.0;
+    double initialVelocity = 0.0;
+
     @Override
     public void setAimAngleRad(double angle) {
-        goalAngRad = angle;
+        if (goalAngleRad != angle) {
+            timer.reset();
+            timer.start();
+
+            initialAngle = sim.getAngleRads();
+            initialVelocity = sim.getVelocityRadPerSec();
+        }
+        goalAngleRad = angle;
```

Comment:
Do we know if this is how the trapezoidal profile will work on the motor controller itself, or will we need to provide the initial conditions to it?

Reply:
I believe it does this on its own.